samples/kotlin_compose/src/androidTest/java/io/customer/android/sample/kotlin_compose/MemoryLeakageInstrumentedTest.kt

```diff
@@ -0,0 +1,243 @@
+
+        // Enter the attribute name and value.
+        val attributeName =
+            composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_attribute_name_input))
+        attributeName.performTextInput("memory test attribute")
+
+        val attributeValue =
+            composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_attribute_value_input))
+        attributeValue.performTextInput("memory test attribute")
+
+        // Click the send button 10
+        for (i in 0..10) {
+            composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_send_device_attribute_button))
+                .performClick()
+        }
+
+        // Assert that there are no leaks.
+        LeakAssertions.assertNoLeaks()
+
+        // Click the back button.
+        composeTestRule.onNodeWithTag(
+            appContext.getString(R.string.acd_back_button_icon),
+            useUnmergedTree = true
+        ).performClick()
+
+        // Click the set profile attribute button.
+        composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_profile_attribute_button))
+            .performClick()
+
+        // Enter the attribute name and value.
+        composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_attribute_name_input))
+            .performTextInput("memory test attribute")
+
+        composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_attribute_value_input))
+            .performTextInput("memory test attribute")
+
+        // Click the send button 10
+        for (i in 0..10) {
+            composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_send_profile_attribute_button))
+                .performClick()
+        }
+
+        // Assert that there are no leaks.
+        LeakAssertions.assertNoLeaks()
+
+        // Click the back button.
+        composeTestRule.onNodeWithTag(
+            appContext.getString(R.string.acd_back_button_icon),
+            useUnmergedTree = true
+        ).performClick()
+
+        // wait till compose ui is idle
+        composeTestRule.runOnIdle {}
+
+        // Click the logout button.
+        composeTestRule.onNodeWithTag(appContext.getString(R.string.acd_logout_button))
+            .performClick()
+
+        composeTestRule.runOnIdle { }
+
+        // Tell the AppWatcher object that we expect the CustomerIO.instance() object to be weakly reachable.
+        AppWatcher.objectWatcher.expectWeaklyReachable(
+            CustomerIO.instance(),
+            "CustomerIO should be garbage collected"
+        )
+
+        // Clear the CustomerIO instance.
+        CustomerIO.clearInstance()
+
+        // Call the System.gc() method to force the garbage collector to run.
+        System.gc()
+
+        // Assert that the AppWatcher object does not have any retained objects.
+        assertEquals(AppWatcher.objectWatcher.hasRetainedObjects, false)
+        assertEquals(AppWatcher.objectWatcher.retainedObjectCount, 0)
+    }
```

Comment:
There is a lot going on in the test function. That's not a bad thing, but I wonder for maintainability purposes. 

If I was to need to modify this test function in the future, what do I need to know about? 

Is the test function long to try and fill up memory? Is it long because it's trying to call many different functions of the SDK?

Reply:
Both, fill out memory and check each part of the app to make sure we know at what point the memory spike came and the possible culprit.